types/materialize-css/common.d.ts

```diff
@@ -30,6 +30,8 @@ declare namespace M {
         options: TOptions;
     }
 
+    type PartialWithNullable<T, K extends keyof T = never> = Partial<Omit<T, K>> & { [ P in K ] ?: T[P] | null };
```

Comment:
I don't like this. What if we need 2 or more keys to be nullable in the future?

Reply:
Oh, maybe we can do `PartialWithNullable<Options, "a" | "b">`. But I still think just declare `defaultDate: Date | null;` is clearer.

Reply:
Yes, this is the intention behind the utility type - any number of keys can be specified via the `K` generic parameter as a union, just wanted to go slow. I am not adamant about that being a part of the PR as it is goal is to provide a way to explicitly pass a nullable type, though. Just think that it can be easier to scale than adding an explicit default. 

Btw, given that defaults for options of `Function` and  `Date` always default to `null` (as far as I know the documentation, but I might be wrong), I can give it a shot and make it more useful (as well as cleaner) by determining nullability based on a union of those types (maybe make it configurable as well). What do you think?

Reply:
I'm fine if you want to go this way. Although I personally would like to just add `| null` to every option that can be null. I think this is clearer to future readers. I would like to hear from others.

Reply:
Sure, thank you - I am fine with either approach. I think the typings would be easier to maintain should the nullability be abstracted away into a generic utility type (not necessarily of the form provided here), though. Let's wait on others to weigh in, and I will go with the flow afterwards by expanding the PR in the chosen direction to cover other components.

Reply:
I agree that adding `| null` to every applicable option would be clearer and more readable than this.

Reply:
@SnowyLeopard understood - I went ahead and switched to explicit nullable options (function signatures look a bit odd as they need to be wrapped in parentheses, maybe something like `Nullable<T> = T | null` could be used for them, but for now I typed them the same way as other options.